### PR TITLE
Arsenal - Add and Improve stats

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Add magnification stat for binoculars.
- Improve magnification stat for optics: shows magnification range of primary optic mode if detected, otherwise minimum and maximum magnification between all optics modes.
- Add vision mode stat for binoculars and NVGs.
- Add night vision generation stat for NVGs.
- Add ammo count stat for magazines.
- Improve conditions on ballistics stats.

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/arsenal/stringtable.xml [hunk 1]
```diff
@@ -1240,6 +1240,70 @@
...
+            <Chinese>視覺模式</Chinese>
+            <Chinesesimp>视觉模式</Chinesesimp>
+            <Korean>보기 모드</Korean>
+            <French>Mode de vision</French>
+            <Polish>Tryb Wizji</Polish>
+            <Russian>Режим видения</Russian>
+            <Portuguese>Modo de Visão</Portuguese>
+            <Czech>Režim sledování</Czech>
+            <Turkish>Görüş Modu</Turkish>
+            <Spanish>Modo de visión</Spanish>
```
[the remark this answers is not in the document]
Comment (LinkIsGrim) [reply addons/arsenal/stringtable.xml:1257]: Snatched from Spectator. Don't know where it's from originally.

File: addons/arsenal/ACE_Arsenal_Stats.hpp [hunk 2]
```diff
@@ -83,6 +91,14 @@ class GVAR(stats) {
         textStatement = QUOTE(call FUNC(statTextStatement_scopeVisionMode));
         tabs[] = {{}, {0}};
     };
+    class ACE_binoVisionMode: statBase {
+        scope = 2;
+        priority = 1.6;
+        displayName = CSTRING(statVisionModeGeneric);
+        showText = 1;
+        textStatement = QUOTE(call FUNC(statTextStatement_binoVisionMode));
+        tabs = {{8,9}, {}};
```
Comment (PabstMirror) [inline addons/arsenal/ACE_Arsenal_Stats.hpp:100]:
```suggestion
        tabs[] = {{8,9}, {}};
```

File: addons/nightvision/ACE_Arsenal_Stats.hpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class EGVAR(arsenal,stats) {
+    class statBase;
+    class GVAR(generation): statBase {
+        scope = 2;
+        priority = 1.6;
+        condition = QUOTE('nvg' in (getArray ((_this select 1) >> 'visionMode') apply {toLower _x}));
+        displayName = CSTRING(NVGeneration);
+        showText = 1;
+        textStatement = QUOTE(call FUNC(statTextStatement_NVGeneration));
+        tabs = {{8}, {}};
```
Comment (PabstMirror) [inline addons/nightvision/ACE_Arsenal_Stats.hpp:10]:
```suggestion
        tabs[] = {{8}, {}};
```

File: addons/arsenal/functions/fnc_statTextStatement_binoVisionMode.sqf [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#include "script_component.hpp"
+/*
+ * Author: Dedmen, johnb43, LinkIsGrim
+ * Text statement for the scope Night vision support stat.
```
Comment (LinkIsGrim) [inline addons/arsenal/functions/fnc_statTextStatement_binoVisionMode.sqf:4]:
```suggestion
 * Text statement for the binocular/NVG vision mode stat.
```

File: addons/nightvision/functions/fnc_statTextStatement_NVGeneration.sqf [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * 1: Item config path <CONFIG>
+ *
+ * Return Value:
+ * Stat Text <STRING>
+ *
+ * Public: No
+ */
+
+params ["", "_config"];
+TRACE_1("statTextStatement_scopeMag",_config);
```
Comment (LinkIsGrim) [inline addons/nightvision/functions/fnc_statTextStatement_NVGeneration.sqf:17]:
```suggestion
TRACE_1("statTextStatement_nvGeneration",_config);
```

File: addons/arsenal/functions/fnc_statTextStatement_binoVisionMode.sqf [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ * 1: Item config path <CONFIG>
+ *
+ * Return Value:
+ * Stat Text <STRING>
+ *
+ * Public: No
+ */
+
+params ["", "_config"];
+TRACE_1("statTextStatement_scopeVisionMode",_config);
```
Comment (LinkIsGrim) [inline addons/arsenal/functions/fnc_statTextStatement_binoVisionMode.sqf:17]:
```suggestion
TRACE_1("statTextStatement_binoVisionMode",_config);
```

File: addons/arsenal/functions/fnc_statTextStatement_binoMag.sqf [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+ * 1: Item config path <CONFIG>
+ *
+ * Return Value:
+ * Stat Text <STRING>
+ *
+ * Public: No
+ */
+
+params ["", "_config"];
+TRACE_1("statTextStatement_scopeMag",_config);
```
Comment (LinkIsGrim) [inline addons/arsenal/functions/fnc_statTextStatement_binoMag.sqf:17]:
```suggestion
TRACE_1("statTextStatement_binoMag",_config);
```

File: addons/arsenal/functions/fnc_statTextStatement_magCount.sqf [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * Return Value:
+ * String to display
+ *
+ * Public: No
+*/
+
+params ["", "_config"];
+
+getNumber (_config >> "count");
```
Comment (LinkIsGrim) [inline addons/arsenal/functions/fnc_statTextStatement_magCount.sqf:18]:
```suggestion
params ["", "_config"];

TRACE_1("statTextStatement_magCount",_config)

getNumber (_config >> "count");
```

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**